### PR TITLE
Add transparent proxy support for OpenBSD

### PR DESCRIPTION
**Pull request by yjh0502**
Add transparent proxy support for OpenBSD with pf divert-to.

### CODE REVIEW

File: docs/transparent/openbsd.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    running as the default gateway.
+
+.. note::
+
+    Note that the **divert-to** rules in the pf.conf given above only apply to
+    inbound traffic. **This means that they will NOT redirect traffic coming
+    from the box running pf itself.** We can't distinguish between an outbound
+    connection from a non-mitmproxy app, and an outbound connection from
+    mitmproxy itself - if you want to intercept your traffic, you should use an
+    external host to run mitmproxy. None the less, pf is flexible to cater for
```
Comment (dwfreed) [inline docs/transparent/openbsd.rst]: "Nonetheless" should be a single word here.

Comment (yjh0502) [inline docs/transparent/openbsd.rst]: @dwfreed I copy-pasted the text from `osx.rst`, so I fixed both documents and pushed.

**Comment by yjh0502** [comment]
Here's some notes.

### implementation
There is other way to implement transparent proxy on OpenBSD, by imitating macOS/FreeBSD implementation by using rdr-to and nat-to rule. Here's pf rule for rdr/nat implementation.
```
pass out on egress proto tcp from $mitm_if to port { 80, 443 } nat-to 127.0.0.1 rdr-to 127.0.0.1 port 8080
```
The rule first NAT outgoing connection to `127.0.0.1` and then redirect traffic to `127.0.0.1:8080` so that mitmproxy can receive the traffic. mitmproxy can inspect destination host by using `pfctl -s state` command (just like how mitmproxy does on osx and FreeBSD) and connect to remote host. This approach has following problems.
 - Recent OpenBSD replaces `sudo` to `doas`, so `platform/osx.py` should be updated to use `doas` on OpenBSD or user should install `sudo` from ports.
 - `mitmproxy/platform/pf.py` should be changed to deal with OpenBSD pfctl output.
 - `pfctl -s state` will be executed on every new connection, which seems quite inefficient.
 - It uses two ports on `127.0.0.1` for each connections, so port space can be exhausted if there are lots of clients/connections. The problem can be avoided by using multiple address and NAT address pools, but it requires additional configuration. 

### `divert-to`
`divert-to` seems to be supported on FreeBSD PF[1], so FreeBSD implementation/documentation can be changed to use `divert-to` rules, which doesn't require `pfctl` to inspect pf states. Unfortunately `divert-to` is not supported on macOS.

[1] https://www.freebsd.org/cgi/man.cgi?query=pf.conf&sektion=5
[2] https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man5/pf.conf.5.html

**Comment by mhils** [comment]
This looks fantastic, thanks! `divert-to` is indeed way better than `pfctl -s state` when available, so I strongly agree with your thoughts.

Thanks! 🎉 🍰 😃

**Pull request merged**